Common/Data/Custom/PsychSignal/PsychSignalSentimentData.cs

```diff
@@ -0,0 +1,134 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+using System;
+using System.Globalization;
+using System.IO;
+
+namespace QuantConnect.Data.Custom.PsychSignal
+{
+    public class PsychSignalSentimentData : BaseData
+    {
+        /// <summary>
+        /// Bullish intensity as reported by psychsignal
+        /// </summary>
+        public decimal BullIntensity;
+
+        /// <summary>
+        /// Bearish intensity as reported by psychsignal
+        /// </summary>
+        public decimal BearIntensity;
+
+        /// <summary>
+        /// Bullish intensity minus bearish intensity
+        /// </summary>
+        public decimal BullMinusBear;
+
+        /// <summary>
+        /// Total bullish scored messages
+        /// </summary>
+        public int BullScoredMessages;
+
+        /// <summary>
+        /// Total bearish scored messages
+        /// </summary>
+        public int BearScoredMessages;
+
+        /// <summary>
+        /// Bull/Bear message ratio.
+        /// </summary>
+        /// <remarks>If bearish messages equals zero, then the resulting value equals zero</remarks>
+        public decimal BullBearMessageRatio;
+
+        /// <summary>
+        /// Total messages scanned.
+        /// </summary>
+        /// <remarks>
+        /// Sometimes, there will be no bull/bear rated messages, but nonetheless had messages scanned.
+        /// This field describes the total fields that were scanned in a minute
+        /// </remarks>
+        public int TotalScoredMessages;
+
+        /// <summary>
+        /// Retrieve Psychsignal data from disk and return it to user's custom data subscription
+        /// </summary>
+        /// <param name="config">Configuration</param>
+        /// <param name="date">Date of this source file</param>
+        /// <param name="isLiveMode">true if we're in livemode, false for backtesting mode</param>
+        /// <returns></returns>
+        public override SubscriptionDataSource GetSource(SubscriptionDataConfig config, DateTime date, bool isLiveMode)
+        {
+            return new SubscriptionDataSource(
+                Path.Combine(
+                    Globals.DataFolder,
+                    "equity",
+                    config.Market,
+                    "alternative",
```

Comment:
Curious as to the decision to use `alternative` in this path. Is this a pattern used elsewhere? The data path formats are structured and there's a small concern this could have unexpected behavior.

@jaredbroad @StefanoRaggi @Martin-Molinero -- can any of you comment here to reassure my potentially rusty brain on the potential implications of this data path format?